README.md

```diff
@@ -1,32 +1,249 @@
+by calling `Thread.currentThread().getName()`.
+In a single threaded program, the default thread is named "main".
+
+The output will be the same every time the program runs.
+
+```text
+Start  (main)
+Snoopy eats treat #1 (main)
+Snoopy eats treat #2 (main)
+Snoopy eats treat #3 (main)
+Snoopy eats treat #4 (main)
+After treats Snoopy wagging is true (main)
+Fifi eats treat #1 (main)
+Fifi eats treat #2 (main)
+Fifi eats treat #3 (main)
+Fifi eats treat #4 (main)
+After treats Fifi wagging is true (main)
+After bath Snoopy wagging is false (main)
+Finish  (main)
+```
+
+Let's use the debugger and visualizer to explore the program and understand the call stack.
+We'll set a breakpoint at the first print statement in the `eatTreats` method.
+The breakpoint is reached after the `main` method calls `snoopy.eatTreats(numTreats)`:
+
+![Single thread stack](https://curriculum-content.s3.amazonaws.com/java-mod-4-threads-intro/introduction-to-threads/singlethread.png)
+
+The `main` method is automatically called when the program starts.
+The call stack contains a frame to store data for the `main` method:
+
+- A local variable that holds a primitive value such as an integer
+  is stored directly in the frame.
+- Objects are not stored in the stack, they are
+  stored in a separate area of memory called the heap. 
+  A local variable that holds an object reference stores the heap address. 
+- Data about the current line of execution is stored in the frame. For example,
+  the `main` method is at line `33`, which corresponds to `snoopy.eatTreats(treats)`.
+
+The `eatTreats` frame will be popped off the stack when the method finishes
+executing its statements. The `main` method can then proceed with its
+next line of code, which will add a new frame to the stack
+corresponding to the call `fifi.eatTreats(treats)`.
+
+Because the single threaded program has just one stack,
+the method calls always execute in the exact order listed:
+
+```java
+snoopy.eatTreats(treats);  
+fifi.eatTreats(treats);    
+snoopy.takeBath();  
+```
+
+## Multithreaded program execution
+
+Let’s briefly look at how we can use multiple threads in Java to support concurrent execution.
+
+Java's `Thread` class provides constructors and methods
+to create and perform operations on a thread.
+
+- The `Thread` class has a constructor that takes a `Runnable` object as an argument.
+- `Runnable` is a functional interface that defines a single abstract method:  `void	run()`
+  - The object passed into the `Thread` constructor must implement the `run()` method.
+- We call the `start()` method on the new `Thread` object.
+  - Starting a thread automatically calls the `run()` method on the `Runnable` object.
+  - The `run()` method executes on the thread's stack.
+
+We will adapt the `main` method to create 3 separate threads to execute
+the `eatTreats` and `takeBath` method calls.  Each thread has its
+own stack, which means the methods can execute is a concurrent rather than
```

Comment:
Typo: "which means the methods can execute is a concurrent rather than sequential order"... do you mean "can execute concurrently rather than in a sequential order"?

Reply:
yes, thanks.